data-mongodb/src/main/resources/META-INF/native-image/io.micronaut.data/micronaut-data-mongodb/native-image.properties

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright 2017-2021 original authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Args = --initialize-at-build-time=org.bson.internal.ProvidersCodecRegistry \
```

Comment:
we need to find out what is initialising these types and not include these flags if we don't have to

Reply:
@radovanradic Can you please take a look at why we need this?

Reply:
Yes, looks like it is needed because of this:
```
package org.bson.conversions;

public interface Bson {
    /**
     * This registry includes the following providers:
     * <ul>
     *     <li>{@link ValueCodecProvider}</li>
     *     <li>{@link BsonValueCodecProvider}</li>
     *     <li>{@link DocumentCodecProvider}</li>
     *     <li>{@link CollectionCodecProvider}</li>
     *     <li>{@link IterableCodecProvider}</li>
     *     <li>{@link MapCodecProvider}</li>
     *     <li>{@link Jsr310CodecProvider}</li>
     *     <li>{@link JsonObjectCodecProvider}</li>
     *     <li>{@link BsonCodecProvider}</li>
     *     <li>{@link EnumCodecProvider}</li>
     * </ul>
     * <p>
     * Additional providers may be added in a future release.
     * </p>
     *
     * @since 4.2
     */
    CodecRegistry DEFAULT_CODEC_REGISTRY =
            fromProviders(asList(
                    new ValueCodecProvider(),
                    new BsonValueCodecProvider(),
                    new DocumentCodecProvider(),
                    new CollectionCodecProvider(),
                    new IterableCodecProvider(),
                    new MapCodecProvider(),
                    new Jsr310CodecProvider(),
                    new JsonObjectCodecProvider(),
                    new BsonCodecProvider(),
                    new EnumCodecProvider()));
```
bunch of these if referred here. When trying to `--initialize-at-run-time=org.bson.conversions.Bson` then get the exception
```
[1/8] Initializing...                                                                                    (0.0s @ 0.17GB)
Error: Incompatible change of initialization policy for org.bson.conversions.Bson: trying to change RUN_TIME from 'META-INF/native-image/io.micronaut.data/micronaut-data-mongodb/native-image.properties' in 'file:///dev/projects/micronaut-data/data-mongodb/build/libs/micronaut-data-mongodb-4.10.0-SNAPSHOT.jar' with 'org.bson.conversions.Bson' to BUILD_TIME interface of org.bson.BsonDocument
```
so looks like we need to have all of these initializations that @sdelamo added.